src/main/java/emissary/output/io/DateFilterFilenameGenerator.java

```diff
@@ -0,0 +1,46 @@
+package emissary.output.io;
+
+import emissary.directory.EmissaryNode;
+import emissary.util.io.FileNameGenerator;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.UUID;
+
+public class DateFilterFilenameGenerator implements FileNameGenerator {
```

Comment:
Don't really love this class name, but it'll do for now.  Should describe the generated filename _(maybe with an example?)_ in a Javadoc comment on the class itself or constructor.

Reply:
I am stuck on a better name. Unless it was going to be really long to be descriptive.
But definitely will add a java doc

Reply:
Again, not in scope for this PR, but we could more filename generators to a child package and then remove the redundant `FilenameGenerator` suffix so the names can have sufficient info without being entire sentences. Alternatively, we could invert the generators so they take a provided NamingStrategy.  (Not sure that last one will always work.)